lib/common/json-schema-validator.js

```diff
@@ -0,0 +1,51 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+module.exports = jsonSchemaValidatorFactory;
+
+jsonSchemaValidatorFactory.$provide = 'JsonSchemaValidator';
+jsonSchemaValidatorFactory.$inject = [
+    'ajv'
+];
+
+function jsonSchemaValidatorFactory(
+    Ajv
+) {
+    function JsonSchemaValidator () {
+        this._ajv = new Ajv({ allErrors: true,  verbose: true});
+        this.addSchema = this._ajv.addSchema;
+        this.addMetaSchema = this._ajv.addMetaSchema;
+    }
+
+    /**
+     * validate JSON data with given JSON schema
+     * @param  {Object|String} schema  JSON schema Object or schema ref id
+     * @param  {Object} data  JSON data to be validated
+     * @return {Boolean} validation pass or throw error
+     */
+    JsonSchemaValidator.prototype.validate = function (schema, data) {
+        if (this._ajv.validate(schema, data)) {
+            return true;
+        } else {
+            var err = new Error('JSON schema validtion failed - ' + this._ajv.errorsText());
+            err.errorList = this._ajv.errors;
+            throw err;
+        }
+    };
+
+    /**
+     * Get compiled schema from the instance by `key` or `ref`.
+     * @param  {String} keyRef `key` that was passed to `addSchema` or 
+     *                  full schema reference (`schema.id` or resolved id).
+     * @return {Object} schema
+     */
+    JsonSchemaValidator.prototype.getSchema = function (key) {
+        var schemaCompiled = this._ajv.getSchema(key);
+        if (schemaCompiled) {
```

Comment:
As investigated, Ajv don't support return a reference resolved schema directly. Additional effort needed to implement a `getSchemaResolved` function.
This will be added in another PR later.